internal/controllers/topology/cluster/structuredmerge/patchhelper.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package structuredmerge
+
+import (
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/cluster-api/internal/contract"
+)
+
+const topologyManager = "topology"
+
+type serverSidePatchHelper struct {
+	client         client.Client
+	modified       *unstructured.Unstructured
+	hasChanges     bool
+	hasSpecChanges bool
+}
+
+// NewServerSidePatchHelper returns a new PatchHelper using server side apply.
+func NewServerSidePatchHelper(scheme *runtime.Scheme, original, modified client.Object, c client.Client, opts ...HelperOption) (PatchHelper, error) {
+	helperOptions := &HelperOptions{}
+	helperOptions = helperOptions.ApplyOptions(opts)
+	helperOptions.allowedPaths = []contract.Path{
```

Comment:
This allowed paths filter out the ownerReference which got created by us.

This results in removing the ownerReference at a second reconciliation.

The old code only adds the ownerReference on creation and does not housekeep it afterwards (But uses it for deletion). 

Some options here (first thoughts, there may be better solutions):
* Solution in code which calls `NewServerSidePatchHelper`: 
  * Add ownerReferences to the allowedPaths here
  * Add the wanted ownerReference always to the modified object
* Solution only in `NewServerSidePatchHelper` (would not require bigger changes in calling code):
  * use managedfields to identify already managed ownerreference and copy it over to the modified object.
  * Maybe optional: add ownerReferences to the allowedPaths here